src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+                }
+            }
+        }
+
+        return super.range(field);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    public int getAlignedDayOfWeekInMonth() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        return ((getDayOfMonth() - 1) % lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    int getAlignedDayOfWeekInYear() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        return ((getDayOfYearAdjusted() - 1) % lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    int getAlignedWeekOfMonth() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        return ((getDayOfMonth() - 1) / lengthOfWeek()) + 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    int getAlignedWeekOfYear() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        return ((getDayOfYear() - 1) / lengthOfWeek()) + 1;
+    }
+
+    /**
+     * Returns the day of the week represented by this date.
+     * <p/>
+     * This returns the ordinal of the enum {@link DayOfWeek}.
+     * Day-of-week do not match those of the ISO calendar system.
+     * In particular, each months starts with Sunday.
+     * Leap-day and year-day are not considered week-days, thus return 0.
+     *
+     * @return the day of the week enumeration: between 1 (Monday) and 7 (Sunday), or 0 (leap-day, year-day)
+     */
+    @Override
+    public int getDayOfWeek() {
+        if (isLeapDay() || isYearDay()) {
+            return 0;
+        }
+
+        int doy = getDayOfYearAdjusted();
+
+        return 1 + ((5 + doy) % 7);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return
+     */
+    @Override
+    long getProlepticMonth() {
+        return getProlepticYear() * lengthOfYearInMonths() + getMonth() - 1;
+    }
+
+    private long getCalculatedProlepticMonth() {
+        return getProlepticYear() * lengthOfYearInMonths() + getCalculatedMonth() - 1;
+    }
+
+    //-------------------------------------------------------------------------
+    @Override  // for covariant return type
+    @SuppressWarnings ("unchecked")
+    public ChronoLocalDateTime<InternationalFixedDate> atTime(final LocalTime localTime) {
+        return (ChronoLocalDateTime<InternationalFixedDate>) ChronoLocalDate.super.atTime(localTime);
+    }
+
+    /**
+     * {@inheritDoc}
+     * @return number of TemporalUnit until given Temporal
+     */
+    @Override
+    public long until(final Temporal endExclusive, final TemporalUnit unit) {
+        InternationalFixedDate end = InternationalFixedDate.from(endExclusive);
+
+        if (unit instanceof ChronoUnit) {
+            switch ((ChronoUnit) unit) {
+                case DAYS:
+                    return daysUntil(end);
+                case WEEKS:
```

Comment:
You need a special `weeksUntil(end)`, because year-day (and to a greater extent, leap-day) throws things off.  Again, see Discordian.